File: GeradorDados/Models/Funcionario.cs
namespace GeradorDados.Models;

public class Funcionario
{
    public int Funcionarioid { get; set; }
    public string Nome { get; set; }
    public DateTime? Datanascimento { get; set; }
    public int? Cargoid { get; set; }
    public int? Empresaid { get; set; }
}

File: api/src/main/java/br/ucsal/clinica/model/Funcionario.java
package br.ucsal.clinica.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


import java.time.LocalDateTime;

@Entity
@Table(name = "funcionario")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Funcionario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private LocalDateTime dataNascimento;

    @OneToOne
    @JoinColumn(name = "cargo_id")
    private Cargo cargo;

    @OneToOne
    @JoinColumn(name = "empresa_id")
    private Empresa empresa;
}
